### ISSUE TITLE
provider/pagerduty: allow "null" as an option for auto_resolve_timeout in pagerduty_service

### ISSUE THREAD

**Issue by EvanKrall** [hashicorp/terraform#9861]
### Terraform Version
`Terraform v0.7.8`

### Affected Resource(s)
- pagerduty_service

### Terraform Configuration Files
```hcl
resource "pagerduty_service" "something" {
    name = "something"
    escalation_policy = "Some escalation policy id"
    acknowledgement_timeout = "${24 * 60 * 60}"
    auto_resolve_timeout = null
}
```

### Expected Behavior
This should set auto_resolve_timeout to null, so as to disable the feature, as described on https://www.terraform.io/docs/providers/pagerduty/r/service.html.

### Actual Behavior
```
Errors:

  * pagerduty_service.something: cannot parse '' as int: strconv.ParseInt: parsing "null": invalid syntax
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by heimweh** [comment]
Hi @EvanKrall. Thanks for reporting this.

It seems like the API allows the value **null** (tested it via https://v2.developer.pagerduty.com/v2/page/api-reference#!/Services/put_services_id) to be passed which would disable the `auto_resolve_timeout` (also affects `acknowledgement_timeout`) but it doesn't seem like the client library support it yet. 

Really sorry about this, will take a look and see if I can get this fixed asap.

A possible work-around for now is to not set the `auto_resolve_timeout` and then manually disable it in the UI (tested this quick and this shouldn't cause a diff in Terraform).

**Comment by paddyforan** [comment]
I've labeled this as an enhancement to track the progress of bringing the feature to Terraform, but in the meantime, is it worth updating the docs to correct the available options?

**Comment by heimweh** [comment]
@paddyforan thanks, absolutely that sounds good :)
Gimme a few minutes and I'll push a doc update.

**Comment by heimweh** [comment]
@paddyforan I pushed a fix for this here (#9881), hopefully that should be enough until we get a proper fix out. Once again, really sorry about this.

**Comment by paddyforan** [comment]
Hey @heimweh, thanks for addressing that so quickly! I've gone ahead and updated the Terraform website with the fix, so it's live now. No worries, things happen. Thanks so much for all your work on the pagerduty provider. I'm gonna actually reopen this issue just so we can track the missing feature, but it's great we've got the accurate docs live now.

**Comment by heimweh** [comment]
Hi everyone,

just wanted to give an update on this since this was fixed in the 0.8.6 release.

**Comment by paddyforan** [comment]
Excellent! Thanks, sorry I missed the PR getting merged. I think we can probably close this now, unless there's something I'm missing. Feel free to chime in/reopen it if you think there's more work to be done here.
